src/main/java/uk/gov/mca/beacons/service/registrations/BeaconsController.java

```diff
@@ -28,8 +31,16 @@ public BeaconsSearchResult findAll() {
     return results;
   }
 
-  @GetMapping(value = "/{hexId}")
-  public Beacon findByHexId(@PathVariable("hexId") String hexId) {
-    return new Beacon();
+  @GetMapping(value = "/{uuid}")
+  public BeaconsSearchResult findByUuid(
+    @PathVariable("uuid") String uuidString
+  ) {
+    final var result = new BeaconsSearchResult();
+    UUID uuid = UUID.fromString(uuidString);
+    Beacon beacon = getAllBeaconsService.find(uuid);
+    List<Beacon> beaconList = new ArrayList<Beacon>();
+    beaconList.add(beacon);
+    result.setBeacons(beaconList);
```

Comment:
if we're continuing along this path I would consider adding a BeaconResult with no size or paging and a single beacon rather than array rather than using the BeaconSearchResult. With this you could also extract the Beacon specific logic out of the BeaconSearchResultSerializer. Maaaaaybe then  BeaconSearchResult { List<BeaconResult> } but that might be too far? 🤔